storage/jellyfish-merkle/src/node_type/mod.rs

```diff
@@ -380,7 +380,7 @@ impl InternalNode {
             let only_child_index = Nibble::from(range_existence_bitmap.trailing_zeros() as u8);
             self.child(only_child_index)
                 .unwrap_or_else(|| {
-                    panic!(
+                    unreachable!(
```

Comment:
This is not unreachable.  This is a legitimately reachable state from which there is no recovery - we have a corrupt tree on storage and cannot continue.

A panic is the best possible option, IMO.  Edit maybe unrecoverable! is the better choice.